process/blocksPool.go

```diff
@@ -90,71 +111,80 @@ func (bp *blocksPool) pruneStorer(index uint64) error {
 }
 
 // PutBlock will put the provided outport block data to the pool
-func (bp *blocksPool) PutBlock(hash []byte, outportBlock *outport.OutportBlock, index uint64) error {
-	shardID := outportBlock.ShardID
-
-	outportBlockBytes, err := bp.marshaller.Marshal(outportBlock)
-	if err != nil {
-		return err
-	}
-
+func (bp *blocksPool) PutBlock(hash []byte, value []byte, newIndex uint64, shardID uint32) error {
 	bp.mutMap.Lock()
 	defer bp.mutMap.Unlock()
 
-	currentIndex, ok := bp.indexesMap[shardID]
+	previousIndex, ok := bp.previousIndexesMap[shardID]
 	if !ok {
-		bp.indexesMap[shardID] = initIndex
-		currentIndex = initIndex
+		bp.previousIndexesMap[shardID] = initIndex
+		previousIndex = initIndex
 	}
 
-	if currentIndex == initIndex {
-		err := bp.storer.Put(hash, outportBlockBytes)
+	if previousIndex == initIndex {
+		err := bp.storer.Put(hash, value)
 		if err != nil {
 			return err
 		}
 
-		bp.indexesMap[shardID] = index
+		bp.previousIndexesMap[shardID] = newIndex
 
 		return nil
 	}
 
-	metaIndex := bp.indexesMap[core.MetachainShardId]
+	isSuccesiveIndex := previousIndex+1 == newIndex
+	if !isSuccesiveIndex {
+		return fmt.Errorf("%w: new index should succesive, previous index %d, new index %d",
```

Comment:
I think the overall error messaging can be improved. I would chain `ErrFailedToPutBlockDataToPool` where `PutBlock` is being called with the error returned by the function itself. For example, I would return at this line only `fmt.Errorof("new index should succesive, previous index %d, new index %d", previousIndex, newIndex)` and in the `dataProcessor` I would chain the ErrFailedToPutBlockDataToPool.